docs/static/persistent-queues.asciidoc

```diff
@@ -5,72 +5,73 @@ WARNING: This functionality is in beta and is subject to change. Deployment in p
 
 By default, Logstash uses in-memory bounded queues between pipeline stages
 (inputs → pipeline workers) to buffer events. The size of these in-memory
-queues is fixed and not configurable. If Logstash terminates abnormally,
-either as the result of a software crash or `kill -9`, it's
-possible to lose queued events. 
+queues is fixed and not configurable. If Logstash terminates abnormally, the
+contents of the in-memory queue will be lost. Such abnormal terminations
+include software crash, `kill -9`, or the host machine loses power.
 
-To prevent event loss in these scenarios, you can configure Logstash to use
-persistent queues. With persistent queues enabled, Logstash stores 
-the queue on disk.
+Temporary machine failures are scenarios where Logstash or its host machine are
+terminated abnormally but are capable of being restarted. To prevent event loss
+in the case of temporary machine failures, you can configure Logstash to use
+persistent queues. With persistent queues enabled, Logstash stores the queue on
+disk.
 
 Persistent queues are also useful for Logstash deployments that require high
-throughput and resiliency. Instead of deploying and managing a message
-broker, such as Redis, RabbitMQ, or Apache Kafka, to handle a mismatch in
-cadence between the shipping stage and the relatively expensive processing
-stage, you can enable persistent queues to buffer events on disk. The persistent queue has configurable sizing, which means that you have more control over
-managing situations that can result in back pressure to the source. See <<backpressure-persistent-queues>>. 
+throughput and durability. Instead of deploying and managing a message broker,
+such as Redis, RabbitMQ, or Apache Kafka, to facilitate a publish-subscriber
+model, you can enable persistent queues to buffer events on disk. 
 
-[[persistent-queues-advantages]]
-==== Advantages of Persistent Queues
+In summary, the two benefits of enabling the persistent queue are as follows:
 
-Using persistent queues:
-
-* Provides protection from losing in-flight messages when the Logstash process is abnormally terminated.
-* Absorb bursts of events without needing an external queueing mechanism like Redis or Kafka.
-* Depending on configuration, can provide an at-least-once message
-delivery guarantee. If Logstash is restarted while events are in-flight,
-Logstash will attempt to deliver messages stored in the persistent queue until
-delivery succeeds at least once. In other words, messages stored in the
-persistent queue may be duplicated, but not lost.
+* Provides protection from in-flight message loss when the Logstash process is abnormally terminated.
+* Absorb bursts of events without needing an external queueing mechanism like Redis or Apache Kafka.
```

Comment:
In 5.1.1 it's not guaranteed because we:

1) `checkpoint.writes: 1` is necessary for such a guarantee and it is not the default.
2) and, we don't fsync the checkpoint file (a bug fixed by #6430)